app/vmalert/alerting.go

```diff
@@ -200,6 +202,19 @@ func (ar *AlertingRule) Exec(ctx context.Context, q datasource.Querier, series b
 	return nil, nil
 }
 
+func expandLabels(m datasource.Metric, labels map[string]string) (map[string]string, error) {
+	metricLabels := make(map[string]string)
+	for _, l := range m.Labels {
+		metricLabels[l.Name] = l.Value
+	}
+	tpl := notifier.AlertTplData{
+		Labels: metricLabels,
+		Value:  m.Value,
+	}
+	// we don't expect `query` templates to work here
```

Comment:
in this case, query func for labels wouldn't work at all. Is it worth to add testcase for it and update docs?

Reply:
Yeah, I was thinking maybe it is a not good idea to allow `query` in labels. However, it is up to user.